src/agoradatatools/reporter.py

```diff
@@ -0,0 +1,130 @@
+    gx_failures: Optional[bool] = field(default=False)
+    gx_failure_message: Optional[str] = field(default=None)
+    adt_output_file: Optional[str] = field(default=None)
+    adt_output_version: Optional[int] = field(default=None)
+    adt_output_link: Optional[str] = field(default=None)
+    data_manifest_file: Optional[str] = field(default=None)
+    data_manifest_version: Optional[int] = field(default=None)
+    data_manifest_link: Optional[str] = field(default=None)
+
+    def set_attributes(self, **kwargs) -> None:
+        """Set attributes for the DatasetReport object.
+
+        Args:
+            **kwargs: Keyword arguments for the DatasetReport object.
+        """
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+    @classmethod
+    def format_link(cls, syn_id: str, version: int) -> str:
+        """Generates a link to a specific version of a synapse entity
+
+        Args:
+            syn_id (str): Synapse ID of the entity.
+            version (int): Version number of the entity.
+
+        Returns:
+            str: Formatted link to the specific version of the entity.
+        """
+        return f"https://www.synapse.org/Synapse:{syn_id}.{version}"
+
+
+@dataclass
+class ADTGXReporter:
+    """
+    Holds all DatasetReport objects needed to update the Synapse table for an ADT run.
+    Includes attributes necessary to execute the upload to the table, as well as data which is common
+    to all DatasetReport objects.
+
+    Attributes:
+        syn: Synapse session object.
+        platform: The platform where the processing was run.
+        run_id: The id of the processing run. This will be passed from the `process` CLI command.
+        table_id: Synapse ID of the Synapse table to be updated.
+        data_manifest_file: Synapse ID of the ADT manifest file.
+        data_manifest_version: Version number of the ADT manifest file.
+        data_manifest_link: URL of the specific version of the ADT manifest file.
+        reports: List of DatasetReport objects to be added to the table.
+    """
+
+    syn: synapseclient.Synapse
+    platform: Platform
+    run_id: str
+    table_id: str
+    data_manifest_file: Optional[str] = field(default=None)
+    data_manifest_version: Optional[int] = field(default=None)
+    data_manifest_link: Optional[str] = field(default=None)
+    reports: Optional[List[DatasetReport]] = field(default_factory=list)
+
+    def add_report(self, report: DatasetReport) -> None:
+        """Adds a DatasetReport object to the list of reports.
+
+        Args:
+            report (DatasetReport): DatasetReport object to be added to the list.
+        """
+        self.reports.append(report)
+
+    def _update_reports_before_upload(self) -> None:
+        """Updates the DatasetReport objects with common attributes before uploading to the table."""
+        timestamp = datetime.datetime.now().strftime("%Y-%m-%dT%H:%M:%S.%fZ")
```

Comment:
This is UTC time and the Synapse tables display UTC time as well AFAIK

Reply:
Ohhh, 
```
datetime.datetime.now().strftime("%Y-%m-%dT%H:%M:%S.%fZ")
```

You're saving this as a string column?  I would convert that to epoch time in milliseconds (numerical) so that synapse can render this as a "date" column.

Reply:
No, it's a date column if you look at the table schema. The Synapse table seems to detect the format of the string timestamp and read it in as a datetime value.

Reply:
Interesting!  I didn't realize Tables supported that too. The more you know...